doc/howto/opam-file-generation.rst

```diff
@@ -0,0 +1,129 @@
+How to generate opam files from ``dune-project``
+================================================
+
+This guide will show you how to configure Dune so that it generates opam files.
+
+Declaring package dependencies
+------------------------------
+
+The goal of this first step is to add ``(package)`` stanzas in your
+``dune-project`` file. These stanzas declare the metadata that your package
+uses, in the language of opam packages. See :ref:`declaring-a-package`.
+
+The next step depends on whether you are starting from a clean slate (new
+package) or adapting an existing opam file.
+
+For a new package (no existing opam file)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If your project does not have any opam files, your will have to find your
+package dependencies. In the simple case, you can collect all the libraries
+that appear in the ``(libraries)`` fields of your project, and put this list in
+the ``(depends)`` field of the corresponding ``(package)``. See
+:doc:`../explanation/ocaml-ecosystem` for the difference between libraries and
+packages.
+
+Example: you have a library that looks like::
+
+  (library
+   (public_name frobnitz)
+   (libraries lwt fmt))
+
+You can declare the package as::
+
+  (package
+   (name frobnitz)
+   (depends lwt fmt)
+
+Also add common metadata using ``(authors)``, ``(maintainers)``, ``(license)``,
+``(source)``, as well as a ``(synopsis)`` and a ``(description)`` for
```

Comment:
Is this sentence complete? .... or does is the "for" to lead into the next heading: "For an Existing Package"?